src/frequenz/channels/broadcast.py

```diff
@@ -249,31 +249,33 @@ def __len__(self) -> int:
         """
         return len(self._q)
 
-    async def receive(self) -> Optional[T]:
-        """Receive a message from the Broadcast channel.
-
-        Waits until there are messages available in the channel and returns
-        them.  If there are no remaining messages in the buffer and the channel
-        is closed, returns `None` immediately.
-
-        If [into_peekable()][frequenz.channels.Receiver.into_peekable] is called
-        on a broadcast `Receiver`, further calls to `receive`, will raise an
-        `EOFError`.
+    async def ready(self) -> None:
+        """Wait until the receiver is ready with a value.
 
         Raises:
-            EOFError: when the receiver has been converted into a `Peekable`.
-
-        Returns:
-            `None`, if the channel is closed, a message otherwise.
+            EOFError: When called before a call to `ready()` finishes.
```

Comment:
Super minimal, but this still needs updating (`ready()` can't be called before `ready()` :D).

Reply:
oh, wow. :D  fixed.